src/core/src/core_logic/PatchInstaller.py

```diff
@@ -162,28 +167,66 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
+            all_packages, all_package_versions, packages, package_versions, maintenance_window, package_manager)
+
+        stopwatch_for_batch_install_process.stop()
+
+        install_update_count_in_batch_patching = installed_update_count
+        attempted_parent_package_install_count_in_batch_patching = self.attempted_parent_package_install_count
+        successful_parent_package_install_count_in_batch_patching = self.successful_parent_package_install_count
+        failed_parent_package_install_count_in_batch_patching = self.failed_parent_package_install_count
+        batch_processing_perf_log = "[{0}={1}][{2}={3}][{4}={5}][{6}={7}][{8}={9}][{10}={11}][{12}={13}][{14}={15}]".format(Constants.PerfLogTrackerParams.TASK, "InstallPackagesInBatches", 
+                                    "InstalledPackagesCountInBatchProcessing", str(install_update_count_in_batch_patching), "AttemptedParentPackageInstallCount", attempted_parent_package_install_count_in_batch_patching,
+                                    "SuccessfulParentPackageInstallCount", successful_parent_package_install_count_in_batch_patching, "FailedParentPackageInstallCount", failed_parent_package_install_count_in_batch_patching,
+                                    "RemainingPackagesToInstall", str(len(packages)), Constants.PerfLogTrackerParams.PATCH_OPERATION_SUCCESSFUL, str(patch_installation_successful), 
+                                    "IsMaintenanceWindowBatchCutoffReached", str(maintenance_window_batch_cutoff_reached))
+
+        stopwatch_for_batch_install_process.write_telemetry_for_stopwatch(str(batch_processing_perf_log))
+
+        if len(packages) == 0:
+            self.log_final_metrics(maintenance_window, patch_installation_successful, maintenance_window_exceeded, installed_update_count)
+            return installed_update_count, patch_installation_successful, maintenance_window_exceeded
+        else:
+            progress_status = self.progress_template.format(str(datetime.timedelta(minutes=maintenance_window.get_remaining_time_in_minutes())), str(self.attempted_parent_package_install_count), str(self.successful_parent_package_install_count), str(self.failed_parent_package_install_count), str(installed_update_count - self.successful_parent_package_install_count),
+                                                        "Following packages are not attempted or failed in batch installation: " + str(packages))
+            self.composite_logger.log(progress_status)
+
+        stopwatch_for_sequential_install_process = Stopwatch(self.env_layer, self.telemetry_writer, self.composite_logger)
+        stopwatch_for_sequential_install_process.start()
 
         for package, version in zip(packages, package_versions):
+            if package not in self.last_still_needed_packages:
+                self.composite_logger.log("The following package is already installed, it could have been installed as dependent package of some other package: " + package)
```

Comment:
Please convert this into a log_debug.

Also write a single log statement after populating a temporary package list inside the for loop.

Blocking comment.

Reply:
Needs to be resolved before deployment. Not holding the PR on this.

Reply:
@rane-rajasi please ensure that this is covered before the next deployment.

Reply:
Created a backlog item for me to change this:#16744995

I will make this change before the next deployment